src/main/java/spireMapOverhaul/zoneInterfaces/EncounterModifyingZone.java

```diff
@@ -0,0 +1,122 @@
+
+import basemod.BaseMod;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.monsters.MonsterGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public interface EncounterModifyingZone {
+    /**
+     * Defines the set of normal encounters for the zone, replacing all normal combats.
+     * If this is non-null, normal encounters in the zone will be randomly selected from this list.
+     * No normal encounter will be fought twice in a row; if only one is defined, the zone will fall back to the
+     * encounters defined by the current act.
+     * @return The list of normal encounters for the zone.
+     */
+    default List<ZoneEncounter> getNormalEncounters() { return null; }
+    /**
+     * Defines the set of elite encounters for the zone, replacing all elite combats.
+     * If this is non-null, elite encounters in the zone will be randomly selected from this list.
+     * No elite encounter will be fought twice in a row; if only one is defined, the zone will fall back to the
+     * encounters defined by the current act.
+     * @return The list of elite encounters for the zone.
+     */
+    default List<ZoneEncounter> getEliteEncounters() { return null; }
+
+    /**
+     * Gets additional monsters that should be added to encounters in the zone (in addition to the enemies already present).
+     * If this is non-null, the monsters will be added to the current encounter.
+     * Note that as part of this method, you can check the current act (use ActUtil.getRealActNum()), the room type
+     * (e.g. AbstractDungeon.getCurrRoom instanceof MonsterRoomElite), etc.
+     * The x position of the monster is automatically determined and will be to the left of all the enemies already present,
+     * with space automatically allocated based on the size of the monster's hitbox. (Any x position set as part of the
+     * monster's constructor will be overwritten.)
+     * Keep the number of size of additional monsters small to avoid them taking up so much space that they overlap with
+     * the player when added to fights that already have many enemies.
+     * @return The monsters to add to the encounter.
+     */
+    default List<AbstractMonster> getAdditionalMonsters() { return null; }
+
+    /**
+     * Defines a new encounter in a zone.
+     * An encounter requires an ID, which should be the same as the monster ID for single enemy encounters. This is
+     * needed by BaseMod (and can be used to fight your new encounter from the console with the fight command).
+     * An encounter also requires an act number, and will only show up in that act.
+     * The enemies in an encounter are defined by a function that returns either a single monster or a MonsterGroup.
+     * An encounter may have a special name. If no name is provided, the encounter will be named based on the enemies.
+     */
+    public static class ZoneEncounter {
```

Comment:
This class is how an encounter is defined.

The base game uses weights so that different encounters show up with different probabilities. That didn't seem necessary for any of the zones in the document, so I left it out. It would be easy to add in as part of the this class.